gcloud-java-compute/src/main/java/com/google/gcloud/compute/MachineType.java

```diff
@@ -0,0 +1,326 @@
+
+    private Builder() {}
+
+    Builder machineTypeId(MachineTypeId machineTypeId) {
+      this.machineTypeId = machineTypeId;
+      return this;
+    }
+
+    Builder id(BigInteger id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder creationTimestamp(String creationTimestamp) {
+      this.creationTimestamp = creationTimestamp;
+      return this;
+    }
+
+    Builder description(String description) {
+      this.description = description;
+      return this;
+    }
+
+    Builder selfLink(String selfLink) {
+      this.selfLink = selfLink;
+      return this;
+    }
+
+    Builder guestCpus(Integer guestCpus) {
+      this.guestCpus = guestCpus;
+      return this;
+    }
+
+    Builder memoryMb(Integer memoryMb) {
+      this.memoryMb = memoryMb;
+      return this;
+    }
+
+    Builder scratchDisks(List<Integer> scratchDisks) {
+      this.scratchDisks = scratchDisks;
+      return this;
+    }
+
+    Builder maximumPersistentDisks(Integer maximumPersistentDisks) {
+      this.maximumPersistentDisks = maximumPersistentDisks;
+      return this;
+    }
+
+    Builder maximumPersistentDisksSizeGb(Long maximumPersistentDisksSizeGb) {
+      this.maximumPersistentDisksSizeGb = maximumPersistentDisksSizeGb;
+      return this;
+    }
+
+    Builder deprecationStatus(DeprecationStatus<MachineTypeId> deprecationStatus) {
+      this.deprecationStatus = deprecationStatus;
+      return this;
+    }
+
+    MachineType build() {
+      return new MachineType(this);
+    }
+  }
+
+  private MachineType(Builder builder) {
+    this.machineTypeId = builder.machineTypeId;
+    this.id = builder.id;
+    this.creationTimestamp = builder.creationTimestamp;
+    this.description = builder.description;
+    this.selfLink = builder.selfLink;
+    this.guestCpus = builder.guestCpus;
+    this.memoryMb = builder.memoryMb;
+    this.scratchDisks = builder.scratchDisks;
+    this.maximumPersistentDisks = builder.maximumPersistentDisks;
+    this.maximumPersistentDisksSizeGb = builder.maximumPersistentDisksSizeGb;
+    this.deprecationStatus = builder.deprecationStatus;
+  }
+
+  /**
+   * Returns the machine type's identity.
+   */
+  public MachineTypeId machineTypeId() {
+    return machineTypeId;
+  }
+
+  /**
+   * Returns an unique identifier for the machine type; defined by the service.
+   */
+  public BigInteger id() {
+    return id;
+  }
+
+  /**
+   * Returns the creation timestamp in RFC3339 text format.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc3339.txt">RFC3339</a>
+   */
+  public String creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns an optional textual description of the machine type.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns a service-defined URL for the machine type.
+   */
+  public String selfLink() {
+    return selfLink;
+  }
+
+  /**
+   * Returns the number of virtual CPUs that are available to the instance.
```

Comment:
I wonder why they qualify this as "guest" but don't do the same for, e.g., memory.